AUTH FEAT: create API for login and register plus little UI.

ahmadbbaihaqi opened the pull request:
Add some library to test this backend. add middleware to protect other page. add model schema for user and thread based on ERD

ahmadbbaihaqi committed:
AUTH FEAT: create API for login and register plus little UI. Add some library to test this backend. add middleware to protect other page. add model schema for user and thread based on ERD

ahmadbbaihaqi committed:
AUTH FIX: resolve conflict merge notif

raihanizar reviewed (changes requested):
**package.json**
- package `next` kehapus, tidak ada Next.js tanpa package `next` wkwk

**prisma/schema.prisma**
- model User ada firstName dan lastName, sedangkan di ERD tidak ada, tolong disesuaikan

**src/app/api/v1/auth/login/route.js**
- Maksud response sudah betul, tapi tolong samakan bentuk response (code & message) dengan api contract yg sudah kita buat.
- Nanti kirimkan JWT_SECRET yg digunakan

**src/app/api/v1/auth/register/route.js**
- Maksud response sudah betul, tapi tolong samakan bentuk response (code & message) dengan api contract yg sudah kita buat.
- Register tidak menerima firstName dan lastName, sesuaikan dengan API contract

raihanizar reviewed (changes requested):
**package.json**
- kayaknya salah resolve conflict, yg sudah ada malah kehapus, 
- kita gak pakai s3 client, karena gak simpan gambar

ahmadbbaihaqi committed:
AUTH FIX : resolve some issue below
**package.json**
- [x] package `next` kehapus. iya nih malah kehapus

**prisma/schema.prisma**
- [x] model User ada firstName dan lastName, sedangkan di ERD tidak ada, tolong disesuaikan

**src/app/api/v1/auth/login/route.js**
- [x] Maksud response sudah betul, tapi tolong samakan bentuk response (code & message) dengan api contract yg sudah kita buat.
- [x] Nanti kirimkan JWT_SECRET yg digunakan

**src/app/api/v1/auth/register/route.js**
- [x] Maksud response sudah betul, tapi tolong samakan bentuk response (code & message) dengan api contract yg sudah kita buat.
- [x] Register tidak menerima firstName dan lastName, sesuaikan dengan API contract

raihanizar reviewed (changes requested):
overall API nya udah jalan dan datanya masuk database, cuma ada beberapa catatan sebelum di-accept

**src/app/api/v1/auth/register/route.js**
- API masih accept firstName, lastName, coba dihilangkan
- skema data response successful belum sesuai dengan API contract: cukup output pesan berhasil aja, jangan semua data apalagi ada password di situ
- catch sebaiknya jangan diberi pesan "email already exist", karena catch buat error general, sedangkan di code ini belum ada logic ngecek email exist / nggak

**src/app/api/v1/auth/register/login.js**
- skema data response successful belum sesuai dengan API contract: atribut `data` coba diganti `payload`, `token` belum ada di dalam response

**src/components/auth/Register.jsx**
- Masih ada input first name last name
- Setelah register, coba bikin redirect ke halaman login

**src/components/auth/Login.jsx**
- Setelah login, coba bikin redirect ke halaman dashboard

ahmadbbaihaqi committed:
AUTH FIX: remove firsname, lastname in route.js and Register.jsx, adjust some response json based on API Contract, add router to login for register and dashboard for login

ahmadbbaihaqi commented:
mohon maaf mas tidak teliti, re check ya mas @raihanizar
